packages/bdd-tests/step_definitions/releaseprescription-steps.ts

```diff
@@ -0,0 +1,97 @@
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[2].resource.identifier[0].value).toBe("9449304130")
+
+    });
+  })
+  test("Release a prescription with an invalid signature", ({ given, when, then, and }) => {
+    ss.givenIAmAuthenticated(given)
+
+    ss.givenICreateXPrescriptionsForSiteWithAnInvalidSignature(given)
+
+    ss.whenIReleaseThePrescription(when)
+
+    then(/^I get no prescription released to (.*)$/, (site) => {
+      expect(ss.resp.data.parameter[1].resource.entry[0].resource.issue[0].details.coding[0].code).toBe("INVALID_VALUE")
+      expect(ss.resp.data.parameter[1].resource.entry[0].resource.issue[0].details.coding[0].display).toBe("Signature is invalid.")
+    });
+
+    and(/^prescription status is (.*)$/, (status) => {
+
+    });
+  });
+
+  test('Release a prescription with multiple line item for a dispensing site', ({ given, when, then, and }) => {
+    ss.givenIAmAuthenticated(given)
+
+    ss.givenICreateXPrescriptionsForSiteWithXLineItems(given)
+
+    ss.whenIReleaseThePrescription(when)
+
+    then(/^I get (\d+) prescription\(s\) released to (.*)$/, (_number, _site) => {
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[0].resource.destination[0].receiver.identifier.value)
+        .toBe(_site)
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[1].resource.resourceType).toBe("MedicationRequest")
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[1].resource.medicationCodeableConcept.coding[0].display)
+        .toBe("Salbutamol 100micrograms/dose inhaler CFC free")
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[1].resource.dispenseRequest.quantity.value).toEqual(200)
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[2].resource.medicationCodeableConcept.coding[0].display)
+        .toBe("Paracetamol 500mg soluble tablets")
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[2].resource.dispenseRequest.quantity.value).toEqual(60)
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[3].resource.medicationCodeableConcept.coding[0].display)
+        .toBe("Methotrexate 10mg/0.2ml solution for injection pre-filled syringes")
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[3].resource.dispenseRequest.quantity.value).toEqual(1)
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[4].resource.medicationCodeableConcept.coding[0].display)
+        .toBe("Flucloxacillin 500mg capsules")
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[4].resource.dispenseRequest.quantity.value).toEqual(28)
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[_number+1].resource.resourceType).toBe("Patient")
+      expect(ss.resp.data.parameter[1].resource.entry[1].resource.entry[_number+1].resource.identifier[0].value).toBe("9449304130")
```

Comment:
we probably shouldn't assume that the medication request will always be the first entry in the bundle

Reply:
I haven't yet seen an example where it isn't. As we haven't sorted out the signature and the response getting in the wrong area, I am happy to leave this as it is now. It can be revisited when we get the response in right area, ie the passedPrescription section in the Json. Explain what you mean by dedup of lookup please